internal/magic/binary.go

```diff
@@ -142,3 +142,131 @@ func Marc(raw []byte, limit uint32) bool {
 	// Field terminator is present.
 	return bytes.Contains(raw, []byte{0x1E})
 }
+
+// CborSeq matches CBOR sequences
+func CborSeq(raw []byte, limit uint32) bool {
+	if len(raw) == 0 {
+		return false
+	}
+	offset, i := 0, 0
+	ok, oldok := true, true
+	for ; ok && offset != len(raw); i++ {
+		oldok = ok
+		offset, ok = cborHelper(raw, offset)
+	}
+	if limit == uint32(len(raw)) {
+		ok = oldok
+	}
+	return ok && i > 1
+}
+
+func cborHelper(raw []byte, offset int) (int, bool) {
+	raw_len := len(raw) - offset
+	if raw_len == 0 {
+		return 0, false
+	}
+
+	mt := uint8(raw[offset] & 0xe0)
+	ai := raw[offset] & 0x1f
+	val := int(ai)
+	offset++
+
+	BgEn := binary.BigEndian
+	switch ai {
+	case 24:
+		if raw_len < 2 {
+			return 0, false
+		}
+		val = int(raw[offset])
+		offset++
+		if mt == 0xe0 && uint64(raw[offset]) < 32 {
+			return 0, false
+		}
+	case 25:
+		if raw_len < 3 {
+			return 0, false
+		}
+		val = int(BgEn.Uint16(raw[offset : offset+2]))
+		offset += 2
+	case 26:
+		if raw_len < 5 {
+			return 0, false
+		}
+		val = int(BgEn.Uint32(raw[offset : offset+4]))
+		offset += 4
+	case 27:
+		if raw_len < 9 {
+			return 0, false
+		}
+		val = int(BgEn.Uint64(raw[offset : offset+8]))
+		offset += 8
+	case 31:
```

Comment:
I am returning `false` here because for those values `well_formed_indefinite` would not return true. But in my code `cborIndefinite` is running `cborHelper` to avoid duplication - this is possible because the `mt` switch in `well_formed_indefinite` and `well_formed` is almost the same. Because of this I exclude the cases that `cborIndefinite` should not handle.